frontend/src/components/projects/LinkSamplesToStudy.tsx

```diff
@@ -0,0 +1,114 @@
+                else return 0
+            })
+        , [projectID, studiesByID])
+    const [study, setStudy] = useState<Study>()
+
+    const [stepsOrder, setStepsOrder] = useState<WorkflowStepOrder[]>([])
+    const [stepOrder, setStepOrder] = useState<WorkflowStepOrder>()
+    useEffect(() => {
+        if (study) {
+            store.dispatch(api.workflows.get(study.workflow_id)).then(response => {
+                setStepsOrder(response.data.steps_order.filter((stepOrder) => stepOrder.order >= study.start && stepOrder.order <= study.end))
+            }).catch(() => store.dispatch(notifyError({
+                id: NOTIFICATION_ID,
+                title: "Failed to get study",
+                description: `Failed to get study ${study}`
+            })))
+        }
+    }, [study])
+
+    return (
+        <Modal
+            title={"Link Samples to Study"}
+            open={open}
+            okButtonProps={{
+                disabled: !(selectedItemIDs.length > 0 && study && stepOrder)
+            }}
+            onOk={() => {
+                if (selectedItemIDs.length > 0 && study && stepOrder) {
+                    store.dispatch(api.projects.addSamplesToStudy(selectedItemIDs, projectID, study.letter, stepOrder.order)).then(
+                        () => {
+                            store.dispatch(notifySuccess({
+                                id: NOTIFICATION_ID,
+                                title: "Samples linked to study",
+                                description: `Successfully linked samples to study ${study.letter} at step "${stepOrder.step_name}"`
+                            }))
+                        },
+                        () => {
+                            store.dispatch(notifyError({
+                                id: NOTIFICATION_ID,
+                                title: "Failed to link samples to study",
+                                description: `Failed to link samples to study ${study.letter} at step "${stepOrder.step_name}"`
+                            }))
+                        })
+                    if (handleOk) {
+                        handleOk()
+                    }
+                }
+            }}
+            onCancel={() => {
+                if (handleCancel) {
+                    handleCancel()
+                }
+            }}
+        >
+            <Form>
+                <Form.Item label={"Study Letter"}>
+                    <Select
+                        onChange={(value) => setStudy(studies.find(study => study.letter === value))}
+                    >
+                        {studies.map(study =>
+                            <Select.Option
+                                key={study.id}
+                                value={study.letter}
+                            >
+                                {study.letter}
+                            </Select.Option>)
+                        }
+                    </Select>
+                    <Select
```

Comment:
Leave some space between the selection fields.

Reply:
Do you have to include both form fields in the same form.item ? The second field would benefit from a label too to make it clear that it defines the target step.